### DISCUSSION TITLE
Version 0.5.1 Beta 1

### DISCUSSION THREAD

**Discussion by DSheirer** [DSheirer/sdrtrunk#1413] [Announcements]
## 1. Features:
* None.  This is a patch release.

## 2. Requirements:
* Windows x86 (64-bit)
* Linux x86/aarch64 (64-bit)
* OSX x86/aarch64 (64-bit)

**Note**: JMBE library must be manually downloaded/built for OSX aarch64 due to OSX sandboxing of downloaded code.
**Note**: Linux for arm-32 is NOT currently supported

## 3. Optional Requirements
**JMBE version 1.0.9+** is required for decoding APCO25 Phase 1/2 and DMR digital audio on Linux and OSX aarch64 and this library can be compiled within the sdrtrunk application.

Legacy instructions for compiling the JMBE library are located at: **https://github.com/DSheirer/sdrtrunk/wiki/JMBE**

## 4. User Manual
* [Wiki](https://github.com/DSheirer/sdrtrunk/wiki)
* [User Manual](https://github.com/DSheirer/sdrtrunk/wiki/User-Manual)

## 5. Support
* [Support](https://github.com/DSheirer/sdrtrunk/wiki/Support)

## 6. What's Changed
* #1397 Decoding Channels Go IDLE After Period of Time by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1407
* #1397 Channel Spectral Display Enhancement by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1406
* #1399 ARS Device Identifier Parsing Issue by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1400
* #1408 Enhance USB Transfer Buffer Error Handling in USB Tuners by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1409
* #1411 Update to Version 0.5.0 beta1 by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1412

**Full Changelog**: https://github.com/DSheirer/sdrtrunk/compare/v0.5.0...v0.5.1-beta1

<hr /><em>This discussion was created from the release <a href='https://github.com/DSheirer/sdrtrunk/releases/tag/v0.5.1-beta1'>Version 0.5.1 Beta 1</a>.</em>

**Comment by Ulti-P-Uszer** [comment]
This release seems much more stable on my 5 SDRs on my 16 core CPU and the ram turned up to 10 gigs in the JVM. It has now been running stably for over an hour.

**Comment by ElroyBJ** [comment]
I am still having issues where it stops and I just see Control there.  Many times its locked up and I have to restart. It might work for an 10 minutes or an hour or two then stop. I have two SDRs Listening to one Site, Apco 25 Phase 1.

**Reply by DSheirer** [reply]
Are there any errors in the application log when it's fully locked up?

**Reply by ElroyBJ** [reply]
Not that I can see.
The second log from today 1/21 I started once without the SDRs plugged in. Then restarted once. It ran for a short time while I was writing this then stopped. I have not shut it down yet in the second log. It shows the control Channel only  and the Spectral displays continue to move until I click the mouse and then those displays stop and the app is locked. To shut it down I have to close the command prompt window.

Thanks

[20230120_sdrtrunk_app.log](https://github.com/DSheirer/sdrtrunk/files/10472507/20230120_sdrtrunk_app.log)
[sdrtrunk_app.log](https://github.com/DSheirer/sdrtrunk/files/10472521/sdrtrunk_app.log)

**Reply by DSheirer** [reply]
There's nothing in the log files that tells me what's going on.

Is this application lockup consistent/reproducible?  Meaning, every time you launch the application it runs fine until you click the mouse?  If so, what area of the screen are you clicking that causes the lockup?  Is it the same area you click on every time?

Do you continue to get audio from the decoding channels while its locked up?

Can you please also post a screen shot of the application when it's in the locked up state?

**Reply by ElroyBJ** [reply]
It does not lock up with clicking until after the audio stops. I am not even using the Laptop at the time just listening. I turn it on to listen while working and then after some time. I no longer hear audio so I look over at it and it looks like this.

![image](https://user-images.githubusercontent.com/123216250/213944004-cf3927b1-03c4-4008-85a4-8b84f62378d6.png)

It does not appear to be locked up. I clicked on Playlist editor and that worked. Then I clicked on events at the bottom and then the display at the top and the rest of the app locked up.

I can minimize it when its locked up and then close the command prompt window which closes both but I can not click the x to close it when its locked up.

The same thing has occurred on the last two versions5.0 Beta 7 and 5.0 release.
I can restart the app without any issue and then the same it could run for 20 minutes or even an hour, but will eventually stop again.

Thanks for helping.
